user/tests/app/input_pinmode/input_pinmode.cpp

```diff
@@ -0,0 +1,122 @@
+#include "Particle.h"
+SYSTEM_MODE(SEMI_AUTOMATIC);
```

Comment:
@technobly App-based tests are fine, but we all know that they are never going to be run. Let's also come up with an automated test that verifies that the pin configuration was unaffected, even if something simple as e.g.

```c++
pinMode(pin, OUTPUT_OPEN_DRAIN_PULLUP);
digitalWrite(pin, 0);
assertEqual(digitalRead(pin), 0);
digitalWrite(pin, 1);
assertEqual(digitalRead(pin), 1);
//
attachInterrupt(pin, ...);
//
digitalWrite(pin, 0);
assertEqual(digitalRead(pin), 0);
digitalWrite(pin, 1);
assertEqual(digitalRead(pin), 1);
```

Reply:
I added a test that checks the pullup/down setting which is the main this affected by this PR.  The INPUT/OUTPUT mode was actually not changed in attachInterrupt() before or now.